builtin/logical/pki/path_tidy.go

```diff
@@ -95,50 +96,69 @@ func (b *backend) pathTidyWrite(
 
 		revokedSerials, err := req.Storage.List("revoked/")
 		if err != nil {
-			return nil, fmt.Errorf("error fetching list of revoked certs: %s", err)
+			return fmt.Errorf("error fetching list of revoked certs: %s", err)
 		}
 
 		var revInfo revocationInfo
 		for _, serial := range revokedSerials {
 			revokedEntry, err := req.Storage.Get("revoked/" + serial)
 			if err != nil {
-				return nil, fmt.Errorf("unable to fetch revoked cert with serial %s: %s", serial, err)
+				return fmt.Errorf("unable to fetch revoked cert with serial %s: %s", serial, err)
 			}
 			if revokedEntry == nil {
-				return nil, fmt.Errorf("revoked certificate entry for serial %s is nil", serial)
+				return fmt.Errorf("revoked certificate entry for serial %s is nil", serial)
 			}
 			if revokedEntry.Value == nil || len(revokedEntry.Value) == 0 {
 				// TODO: In this case, remove it and continue? How likely is this to
 				// happen? Alternately, could skip it entirely, or could implement a
 				// delete function so that there is a way to remove these
-				return nil, fmt.Errorf("found revoked serial but actual certificate is empty")
+				return fmt.Errorf("found revoked serial but actual certificate is empty")
 			}
 
 			err = revokedEntry.DecodeJSON(&revInfo)
 			if err != nil {
-				return nil, fmt.Errorf("error decoding revocation entry for serial %s: %s", serial, err)
+				return fmt.Errorf("error decoding revocation entry for serial %s: %s", serial, err)
 			}
 
 			revokedCert, err := x509.ParseCertificate(revInfo.CertificateBytes)
 			if err != nil {
-				return nil, fmt.Errorf("unable to parse stored revoked certificate with serial %s: %s", serial, err)
+				return fmt.Errorf("unable to parse stored revoked certificate with serial %s: %s", serial, err)
 			}
 
 			if time.Now().After(revokedCert.NotAfter.Add(bufferDuration)) {
 				if err := req.Storage.Delete("revoked/" + serial); err != nil {
-					return nil, fmt.Errorf("error deleting serial %s from revoked list: %s", serial, err)
+					return fmt.Errorf("error deleting serial %s from revoked list: %s", serial, err)
 				}
 				tidiedRevoked = true
 			}
 		}
 
 		if tidiedRevoked {
 			if err := buildCRL(b, req); err != nil {
-				return nil, err
+				return err
 			}
 		}
 	}
 
+	// Unset flag
+	atomic.StoreInt32(&b.tidyRunning, 0)
```

Comment:
I think that this should be a `defer` statement.

Reply:
Otherwise, if a running tidy operation errors out, we won't reset the `tidyRunning` and may never be able to run tidy again.